apps/next/app/api/deploy/up/route.ts

```diff
@@ -0,0 +1,46 @@
+import { clerkClient } from "@clerk/nextjs";
+import { type NextRequest } from "next/server";
+import { createWriteStream, existsSync, mkdirSync } from "node:fs";
+import { extract } from "tar";
+
+export const dynamic = "force-dynamic";
+
+export async function POST(request: NextRequest) {
+  const authStatus = await clerkClient.authenticateRequest({ request });
+  if (!authStatus.isSignedIn) {
+    return new Response(JSON.stringify({}), {
+      status: 401,
+    });
+  }
+
+  if (!request.body) {
+    return new Response(JSON.stringify({}), { status: 400 });
+  }
+
+  const tempDir = 'temp';
+  if (!existsSync(tempDir)) {
+    mkdirSync(tempDir);
+  }
+
+  const tag = `foo-${Date.now()}`;
+  const filename = `${tag}.gz`;
+  const uploadedTarball = createWriteStream(filename, "utf8");
+  const extractor = extract({
+    C: tempDir,
+  });
+
+  const writableUploadStream = new WritableStream<Uint8Array>({
+    write(chunk) {
+      uploadedTarball.write(chunk);
+      extractor.write(Buffer.from(chunk));
+    },
+  });
+
+  await request.body?.pipeTo(writableUploadStream);
+  uploadedTarball.close();
```

Comment:
this endpoint is also super clean and mostly web APIs! (`uploadedTarball` being the exception... maybe https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/tee can split the request body readable stream into two readable streams, then use one of those like this https://bun.sh/guides/write-file/stream to write to the upload tarball and the other you can `pipeTo` the custom writableUploadStream to handle the extractor.write stuff)

Reply:
Ah I see. Essentially killing two birds with one stone. I'll give it a shot.

Reply:
I tried it out and ran into these issues:
- trying to call `Bun.write()` all failed with errors like "can't call write on an undefined object". I tried using these all to no avail:
  - Bun.write()
  - import { write } from "bun"
  - globalThis.Bun.write()
  - globalThis.write()
  - global.Bun.write()
  - global.write()
- Using just one of the tee'd/branched streams to save the .gz file didn't work either. Might be to the backed up data that never got read or something else I don't know.

Reply:
You can take a look at commit # https://github.com/onmetal-dev/metal/pull/16/commits/c8cc48c25f7a0ed610dc99b366f061c54705d0b4 for details.

Reply:
IDK if the `tar` command will work on Windows, and frankly I don't think so. Do you know?

Reply:
All done! Also removed the `tar` package from the Next app due to the change above rendering it unneeded.